metsim/metsim.py

```diff
@@ -133,34 +141,43 @@ class MetSim(object):
         "lapse_rate": 0.0065,
         "iter_dims": ['lat', 'lon'],
         "out_vars": ['temp', 'prec', 'shortwave', 'longwave',
-                     'vapor_pressure', 'rel_humid']
+                     'vapor_pressure', 'rel_humid'],
+        "chunks": NO_SLICE,
+        "scheduler": 'single-threaded',
+        "num_workers": 1
     }
 
-    def __init__(self, params: dict):
+    def __init__(self, params: dict, domain_slice=NO_SLICE):
         """
         Constructor
         """
+        self._domain = None
+        self._met_data = None
+        self._state = None
+        self._client = None
+        self._domain_slice = domain_slice
+
         # Record parameters
         self.params.update(params)
 
-        logger.setLevel(self.params['verbose'])
-        ch.setLevel(self.params['verbose'])
-        logger.addHandler(ch)
-        self._validate_setup()
-        logger.info("read_domain")
-        self.domain = io.read_domain(self.params)
+        # set global dask scheduler
+        if domain_slice is NO_SLICE:
+            if 'distributed' == self.params['scheduler']:
+                from distributed import Client
+                self._client = Client(
+                    n_workers=self.params['num_workers'], threads_per_worker=1, diagnostics_port=5555)
+                print(self._client)
```

Comment:
if scheduler is a file, we should pass scheduler to `Client(sheduler_file=scheduler)`.